mlir/test/Dialect/LLVMIR/inlining.mlir

```diff
@@ -56,3 +56,122 @@ func.func @test_not_inline(%ptr : !llvm.ptr) -> () {
   call @with_mem_attr(%ptr) : (!llvm.ptr) -> ()
   return
 }
+
+// -----
+
+// Include all function attributes that don't prevent inlining
+llvm.func internal @callee() -> (i32) attributes { function_entry_count = 42 : i64, dso_local } {
+  %0 = llvm.mlir.constant(42 : i32) : i32
+  llvm.return %0 : i32
+}
+
+// CHECK-LABEL: llvm.func @caller
+// CHECK-NEXT: %[[CST:.+]] = llvm.mlir.constant
+// CHECK-NEXT: return %[[CST]]
+llvm.func @caller() -> (i32) {
+  // Include all call attributes that don't prevent inlining.
+  %0 = llvm.call @callee() { branch_weights = dense<42> : vector<1xi32>, fastmathFlags = #llvm.fastmath<nnan, ninf> } : () -> (i32)
+  llvm.return %0 : i32
+}
+
+// -----
+
+llvm.func @foo() -> (i32) attributes { passthrough = ["noinline"] } {
+  %0 = llvm.mlir.constant(0 : i32) : i32
+  llvm.return %0 : i32
+}
+
+llvm.func @bar() -> (i32) attributes { passthrough = ["noinline"] } {
+  %0 = llvm.mlir.constant(1 : i32) : i32
+  llvm.return %0 : i32
+}
+
+llvm.func @callee_with_multiple_blocks(%cond: i1) -> (i32) {
+  llvm.cond_br %cond, ^bb1, ^bb2
+^bb1:
+  %0 = llvm.call @foo() : () -> (i32)
+  llvm.br ^bb3(%0: i32)
+^bb2:
+  %1 = llvm.call @bar() : () -> (i32)
+  llvm.br ^bb3(%1: i32)
+^bb3(%arg: i32):
+  llvm.return %arg : i32
+}
+
+// CHECK-LABEL: llvm.func @caller
+// CHECK-NEXT: llvm.cond_br {{.+}}, ^[[BB1:.+]], ^[[BB2:.+]]
+// CHECK-NEXT: ^[[BB1]]:
+// CHECK-NEXT: llvm.call @foo
+// CHECK-NEXT: llvm.br ^[[BB3:[a-zA-Z0-9_]+]]
+// CHECK-NEXT: ^[[BB2]]:
+// CHECK-NEXT: llvm.call @bar
+// CHECK-NEXT: llvm.br ^[[BB3]]
+// CHECK-NEXT: ^[[BB3]]
+// CHECK-NEXT: llvm.br ^[[BB4:[a-zA-Z0-9_]+]]
+// CHECK-NEXT: ^[[BB4]]
+// CHECK-NEXT: llvm.return
+llvm.func @caller(%cond: i1) -> (i32) {
+  %0 = llvm.call @callee_with_multiple_blocks(%cond) : (i1) -> (i32)
+  llvm.return %0 : i32
+}
+
+// -----
+
+llvm.func @personality(...) -> i32
```

Comment:
Just wondering, must a personality function be variadic?

Reply:
I made it non-variadic 🤷‍♂️